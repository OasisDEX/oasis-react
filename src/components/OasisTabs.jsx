import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import { NavLink } from 'react-router-dom';
import styles from './OasisTabs.scss';
import CSSModules from 'react-css-modules';

const propTypes = PropTypes && {
  pathname: PropTypes.string,
  activeTradingPair: PropTypes.object,
  defaultTradingPair: PropTypes.object.isRequired,
};
const defaultProps = {};

class OasisTabs extends PureComponent {
  isActive(basePath) {
    const  { pathname } = this.props;
    return (pathname && pathname.indexOf(basePath) === 0) ? styles.active : ''
  }
  render() {
    const  { defaultTradingPair, activeTradingPair } = this.props;
    const { baseToken, quoteToken } = activeTradingPair ? activeTradingPair : defaultTradingPair;
    return (
      <div className={styles.rowTabs}>
        <ul className={styles.tabs}>
          <li className={`${styles.tab} ${this.isActive('/trade')}`}>
            <NavLink to={`/trade/${baseToken}/${quoteToken}`}>Trade</NavLink>
          </li>
          <li className={`${styles.tab} ${styles.tabBig} ${this.isActive('/wrap-unwrap')}`}>
            <NavLink to={'/wrap-unwrap'}>Wrap / Unwrap</NavLink>
          </li>
          <li className={`${styles.tab} ${this.isActive('/transfer')}`}>
            <NavLink to={'/transfer'}>Transfer</NavLink>
          </li>
        </ul>
      </div>
    );
  }
}

OasisTabs.displayName = 'OasisTabs';
OasisTabs.propTypes = propTypes;
OasisTabs.defaultProps = defaultProps;
export default CSSModules(OasisTabs, styles, {allowMultiple: true});
