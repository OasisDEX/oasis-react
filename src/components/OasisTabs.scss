@import '../styles/modules/constants';
@import '../styles/modules/grid';
@import '../styles/modules/labels';

.row-tabs {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 60px 0;


  &:before , &:after{
    content: '';
    width: calc((100% - #{$tabs-width}) / 2);
    border-top:0.5px solid $gray;
    border-bottom:0.5px solid $gray;
  }
}

.Tabs {
  display: flex;

  width: $tabs-width;

  border: 1px solid $gray;
  border-radius: 100px;

  list-style-type: none;
  padding-left: 0;
  margin-bottom: 0;

  overflow: hidden;

  & .Tab {
    flex-basis: 100%;
    height: 40px;
    border-right: 1px solid $gray;

    text-align: center;
    text-transform: uppercase;
    letter-spacing: 1px;

    &--2x {
      flex-basis: 200%;
    }

    &:hover {
      background: $tabs-background--on-hover;
      cursor: pointer;

      & a {
        color: $gray-blueish;
      }
    }

    &:last-child {
      border-right: 0;
    }

    & a {
      color: $gray;
      text-decoration: none;
      @include text();
      width: 100%;
      height: 100%;
      display: flex;
      justify-content: center;
      align-items: center;

      &:hover {
        color: $gray-blueish;
      }
    }
  }
}