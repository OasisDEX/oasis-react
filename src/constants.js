/**
 *   NETWORKS
 */
export const MAIN_NET_ID = "1";
export const KOVAN_NET_ID = "42";
export const KOVAN = "kovan";
export const MAIN = "main";

/**
 * NETWORK STATUS
 */
export const ONLINE = "NETWORK/ONLINE";
export const CONNECTING = "NETWORK/CONNECTING";
export const OUT_OF_SYNC = "NETWORK/OUT_OF_SYNC";
export const CLOSED = "NETWORK/CLOSED";

/**
 * BIG NUMBER
 */
export const BN_DECIMAL_PRECISION = 18;

/**
 * PRECISION
 */

export const ETH_UNIT_WEI = "wei"; // wei 	1 wei 	1
export const ETH_UNIT_KWEI = "kwei"; // Kwei (babbage) 	1e3 wei 	1,000
export const ETH_UNIT_MWEI = "mwei"; // Mwei (lovelace) 	1e6 wei 	1,000,000
export const ETH_UNIT_GWEI = "gwei"; // Gwei (shannon) 	1e9 wei 	1,000,000,000
export const ETH_UNIT_MICROETHER = "microether"; // microether (szabo) 	1e12 wei 	1,000,000,000,000
export const ETH_UNIT_MILIETHER = "milliether"; // milliether (finney) 	1e15 wei 	1,000,000,000,000,000
export const ETH_UNIT_ETHER = "ether"; // ether 	1e18 wei 	1,000,000,000,000,000,000

export const HAS_ACCOUNTS = true;
export const NO_ACCOUNTS = false;

export const TOKEN_ETHER = "ETH";
export const TOKEN_WRAPPED_ETH = "W-ETH";
export const TOKEN_WRAPPED_GNT = "W-GNT";
export const TOKEN_DAI = "DAI";
export const TOKEN_MAKER = "MKR";
export const TOKEN_DIGIX = "DGD";
export const TOKEN_GOLEM = "GNT";
export const TOKEN_RHOC = "RHOC";

export const QUOTE_TOKENS = [TOKEN_WRAPPED_ETH, TOKEN_DAI];

export const BASE_TOKENS = [
  TOKEN_DAI,
  TOKEN_WRAPPED_GNT,
  TOKEN_DIGIX,
  TOKEN_RHOC,
];

export const MAKE_BUY_OFFER = "OFFER_MAKES/MAKE_BUY_OFFER";
export const MAKE_SELL_OFFER = "OFFER_MAKES/MAKE_SELL_OFFER";
export const MAKE_SELL_OFFER_FORM_NAME = "makeSellOffer";
export const MAKE_BUY_OFFER_FORM_NAME = "makeBuyOffer";

export const TOKEN_ALLOWANCE_TRUST_STATUS_DISABLED_MIN_MAX = 0;
export const TOKEN_ALLOWANCE_TRUST_STATUS_ENABLED = true;
export const TOKEN_ALLOWANCE_TRUST_STATUS_DISABLED = false;
export const TOKEN_ALLOWANCE_TRUST_STATUS_NOT_SET = undefined;
export const TOKEN_ALLOWANCE_TRUST_STATUS_LOADING = null;
export const TOKEN_ALLOWANCE_TRUST_STATUS_ENABLED_MIN =
  "0xffffffffffffffffffffffffffffffff";
export const TOKEN_ALLOWANCE_TRUST_STATUS_ENABLED_MAX =
  "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";

export const TOKEN_ALLOWANCE_TRUST_SUBJECT_TYPE_MARKET =
  "BALANCES/TOKEN_ALLOWANCE_TRUST_SUBJECT_TYPE_MARKET";
export const TOKEN_ALLOWANCE_TRUST_SUBJECT_TYPE_ADDRESS =
  "BALANCES/TOKEN_ALLOWANCE_TRUST_SUBJECT_TYPE_ADDRESS";

export const SYNC_STATUS_PRISTINE = "SYNC_STATUS_PRISTINE";
export const SYNC_STATUS_PENDING = "SYNC_STATUS_PENDING";
export const SYNC_STATUS_COMPLETED = "SYNC_STATUS_COMPLETED";
export const SYNC_STATUS_ERROR = "SYNC_STATUS_ERROR";

export const SETMAXBTN_HIDE_DELAY_MS = 100;

/**
 *  GAS and GAS_PRICE default values
 */
export const DEFAULT_GAS_LIMIT = "1000000"; // 1M GAS
export const DEFAULT_GAS_PRICE = "6000000000"; // 6 GWEI

/**
 * Subscriptions
 */

export const SUBSCRIPTIONS_GROUP_GLOBAL_INITIAL = 'SUBSCRIPTIONS/GROUP_GLOBAL/INITIAL';
export const SUBSCRIPTIONS_GROUP_ACCOUNT_SPECIFIC_INITIAL = 'SUBSCRIPTIONS/GROUP_ACCOUNTS_SPECIFIC/INITIAL';

export const SUBSCRIPTIONS_LATEST_BLOCK = "SUBSCRIPTIONS/LATEST_BLOCK";
export const SUBSCRIPTIONS_TOKEN_BALANCE_EVENTS = "SUBSCRIPTIONS/TOKEN_BALANCES";
export const SUBSCRIPTIONS_ORDERS_EVENTS = "SUBSCRIPTIONS/ORDERS_EVENTS";
export const SUBSCRIPTIONS_ETHER_BALANCE_CHANGE_EVENTS = "SUBSCRIPTIONS/ETHER_BALANCE_CHANGE_EVENTS";
export const SUBSCRIPTIONS_LOG_TAKE_EVENTS = "SUBSCRIPTIONS/LOG_TAKE_EVENTS";
export const SUBSCRIPTIONS_USER_LOG_TAKE_EVENTS = "SUBSCRIPTIONS/USER_LOG_TAKE_EVENTS";
export const SUBSCRIPTIONS_TOKEN_TRANSFER_EVENTS = "SUBSCRIPTIONS/TOKEN_TRANSFER_EVENTS";
export const SUBSCRIPTIONS_TOKEN_TRANSFER_HISTORY_EVENTS = "SUBSCRIPTIONS/TOKEN_TRANSFER_HISTORY_EVENTS";
export const SUBSCRIPTIONS_TOKEN_WRAP_UNWRAP_HISTORY_EVENTS = "SUBSCRIPTIONS/TOKEN_TRANSFER_HISTORY_EVENTS";


export const subscriptionGroupToKeyMap = {
  [SUBSCRIPTIONS_GROUP_GLOBAL_INITIAL]: 'globalInitial',
  [SUBSCRIPTIONS_GROUP_ACCOUNT_SPECIFIC_INITIAL]: 'accountSpecificInitial'
};

export const subscriptionTypeToKeyMap = {
  [SUBSCRIPTIONS_LATEST_BLOCK]: 'latestBlock',
  // [SUBSCRIPTIONS_TOKEN_BALANCE_EVENTS]: 'tokenBalances',
  [SUBSCRIPTIONS_ORDERS_EVENTS]: 'ordersEvents',
  [SUBSCRIPTIONS_LOG_TAKE_EVENTS]: 'latestBlock',
  [SUBSCRIPTIONS_TOKEN_WRAP_UNWRAP_HISTORY_EVENTS]: 'wrapUnwrapHistory',
  [SUBSCRIPTIONS_ETHER_BALANCE_CHANGE_EVENTS]: 'etherBalance',
  [SUBSCRIPTIONS_TOKEN_TRANSFER_EVENTS]: 'tokenTransfers'
};

/**
 * Validation errors
 */

export const VALIDATION_VALUE_IS_REQUIRED = 'VALIDATION/VALUE_IS_REQUIRED';

/**
 * Global precision - 10**18
 * @type {number}
 */
export const APP_BASE_PRECISION = 18;

export const USER_TO_LOG_TAKE_OFFER_RELATION_TAKEN_BY_USER = 'USER_TO_LOG_TAKE_RELATION/TAKEN_BY_USER';
export const USER_TO_LOG_TAKE_OFFER_RELATION_USER_MADE = 'USER_TO_LOG_TAKE_RELATION/USER_MADE';
export const USER_TO_LOG_TAKE_OFFER_RELATION_NONE = 'USER_TO_LOG_TAKE_RELATION/NONE';

export const PROVIDER_TYPE_METAMASK = 'PROVIDER_TYPE/METAMASK';
export const PROVIDER_TYPE_MIST = 'PROVIDER_TYPE/MIST';
