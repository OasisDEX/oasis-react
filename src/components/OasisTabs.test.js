/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';

import OasisTabs from './OasisTabs';
import { shallow } from 'enzyme';
import { TOKEN_DAI, TOKEN_WRAPPED_ETH } from '../constants';

describe('(Component) OasisTabs', () => {
  it('should render', () => {
    const props = {
      defaultTradingPair: {
        baseToken: TOKEN_WRAPPED_ETH,
        quoteToken: TOKEN_DAI
      }
    };
    const wrapper = shallow(
      <OasisTabs {...props}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
