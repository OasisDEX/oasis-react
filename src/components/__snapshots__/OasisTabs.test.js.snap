// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisTabs should render 1`] = `
<div
  styleName="row-tabs"
>
  <ul
    styleName="Tabs"
  >
    <li
      styleName="Tab"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/trade"
      >
        Trade
      </NavLink>
    </li>
    <li
      styleName="Tab Tab--2x"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/wrap-unwrap"
      >
        Wrap / Unwrap
      </NavLink>
    </li>
    <li
      styleName="Tab"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/transfer"
      >
        Transfer
      </NavLink>
    </li>
  </ul>
</div>
`;
